src/tests/test_prep.py

```diff
@@ -6,6 +6,129 @@
 import hivpy.column_names as col
 from hivpy.common import SexType, date, timedelta
 from hivpy.population import Population
+from hivpy.prep import PrEPType
+
+
+def reset_prep_willingness_cols(pop: Population):
+    pop.set_present_variable(col.PREP_ORAL_PREF, 0)
+    pop.set_present_variable(col.PREP_CAB_PREF, 0)
+    pop.set_present_variable(col.PREP_LEN_PREF, 0)
+    pop.set_present_variable(col.PREP_VR_PREF, 0)
+    pop.set_present_variable(col.PREP_PREF_RANKED, None)
+    pop.set_present_variable(col.PREP_ORAL_WILLING, False)
+    pop.set_present_variable(col.PREP_CAB_WILLING, False)
+    pop.set_present_variable(col.PREP_LEN_WILLING, False)
+    pop.set_present_variable(col.PREP_VR_WILLING, False)
+    pop.set_present_variable(col.PREP_ANY_WILLING, False)
+
+
+def test_prep_willingness():
+    N = 100
+    pop = Population(size=N, start_date=date(1999, 1, 1))
+    pop.data[col.AGE] = 20
```

Comment:
```suggestion
    pop.data[col.AGE] = [10, 20] * (N / 2)
```
This test would be more effective if some people were _under_ 15 and therefore we checked that it was only updating for over 15s